ipatests/test_integration/test_dns_locations.py

```diff
@@ -25,15 +27,47 @@
     (DNSName(u'_kpasswd._udp'), 464),
 )
 
+IPA_DEFAULT_ADTRUST_SRV_REC = (
+    # srv record name, port
+    (DNSName(u'_ldap._tcp.Default-First-Site-Name._sites.dc._msdcs'), 389),
+    (DNSName(u'_ldap._tcp.dc._msdcs'), 389),
+    (DNSName(u'_kerberos._tcp.Default-First-Site-Name._sites.dc._msdcs'), 88),
+    (DNSName(u'_kerberos._udp.Default-First-Site-Name._sites.dc._msdcs'), 88),
+    (DNSName(u'_kerberos._tcp.dc._msdcs'), 88),
+    (DNSName(u'_kerberos._udp.dc._msdcs'), 88),
+)
+
+# we keep NTP SRV record separated as NTP is "optional"
+IPA_DEFAULT_NTP_SRV_REC = (
+    # srv record name, port
+    (DNSName("_ntp._udp"), 123),
+)
+
+IPA_CA_A_REC = (
+    (DNSName(unicode(IPA_CA_RECORD))),
+)
+
 
 def resolve_records_from_server(rname, rtype, nameserver):
     res = dns.resolver.Resolver()
     res.nameservers = [nameserver]
-    res.lifetime = 10
-    logger.debug("Query: %s %s, nameserver %s", rname, rtype, nameserver)
-    ans = res.query(rname, rtype)
-    logger.debug("Answer: %s", ans.rrset)
-    return ans.rrset
+    res.lifetime = 30
+    logger.info("Query: %s %s, nameserver %s", rname, rtype, nameserver)
+    # lets try to query 3x
+    for i in range(3):
+        try:
+            ans = res.query(rname, rtype)
+            break
+        except (dns.resolver.NXDOMAIN, dns.resolver.Timeout) as e:
+            ans = e
+            time.sleep(10)
+            pass
+    try:
+        logger.info("Answer: %s", ans.rrset)
+        return ans.rrset
+    except AttributeError:
+        pytest.fail("Query: %s %s, nameserver %s failed due {}",
```

Comment:
if you use `return ans....` in the previous try block, you don't need this try/except block. Anything here is error so just fail:
```
pytest.fail("Query: %s %s, nameserver %s failed due {}", rname, rtype, nameserver, error)
```

(Also instead of catching AttributeError would be better to use `isinstance(ans, Exception)` to detect exception)